crates/red_knot_python_semantic/resources/mdtest/unpacking.md

```diff
@@ -93,7 +92,7 @@ reveal_type(d)  # revealed: Unknown
 ### Starred expression (2)
 
 ```py
-[a, *b, c] = (1, 2)  # error: "Object of type `None` is not iterable"
+[a, *b, c] = (1, 2)
```

Comment:
I'm having trouble tracing this -- where is this diagnostic even coming from today?

Reply:
What's the issue? Is this failing? I think that's probably coming from `infer_starred_expression`

Reply:
The diagnostic is no longer showing up, so I had to remove these assertions along with the TODOs to remove them (?).

Reply:
Yeah, I think that's correct. The diagnostics shouldn't be present in these cases and I think I left out one TODO in this specific case which might've confused you (?). The reason these diagnostics were showing up in the first place is because red knot tries to infer the starred expression (`*b`) but that isn't supported yet, before this PR it would've returned `None` and then it would raise a "not-an-iterable" diagnostic. But, it's now changed to use `Never` in this PR.

https://github.com/astral-sh/ruff/blob/c6b82151dddf842039aad72051a942c47e0759dc/crates/red_knot_python_semantic/src/types/infer.rs#L2341-L2344

Reply:
This PR shouldn't even affect this case (we should never use the type of a Store-context expression), but it does just because we don't support assigning to Starred yet, and the way the temporary code works out, it does currently try to iterate the inferred type of a Store-context expression. But not throwing the error is better, so if anything switching from `None` to `Never` is an improvement here.